stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,217 @@
+                            page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+                        else:
+                            page_size = remaining_records
+                    # Handling expired token
+                    elif response_wrapper.code == 401 and not token_generated and \
+                            (response == '' or
+                             'Signature has expired' in response_dict.get('error', {}).get('message', '')):
+                        jwt_token = await self.get_token()
+                        token_generated = True
+                        continue
+                    else:
+                        error = response_dict.get('error', '')
+                        if response_wrapper.code == 403:
+                            error = 'Query length is too long or Invalid Query'
+                        return_obj = self.exception_response(response_wrapper.code, error)
+                        data = []
+                        break
+
+                if data:
+                    data = self.get_results_data(data)
+                    if metadata:
+                        return_obj['data'] = data if data else []
+                    else:
+                        return_obj['data'] = data[int(offset): total_records] if data else []
+                    if jwt_token and next_url and result_count < self.api_client.result_limit:
+                        return_obj['metadata'] = {"result_count": result_count, "jwtToken": jwt_token,
+                                                  "next_url": next_url}
+                else:
+                    if not return_obj.get('error') and return_obj.get('success') is not False:
+                        return_obj['success'] = True
+                        return_obj['data'] = []
+            else:
+                return_obj['success'] = True
+                return_obj['data'] = []
+
+        except InvalidMetadataException as ex:
+            return_obj = self.exception_response(422, f'Invalid metadata: {str(ex)}')
+
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: return_object, dict
+        """
+        return_obj = {}
+        try:
+            token = await self.get_token()
+            response = await self.api_client.ping_data_source(token)
+            response_code = response.code
+            response_dict = json.loads(response.read().decode('utf-8'))
+            if response_code == 200:
+                return_obj['success'] = True
+            else:
+                return_obj = self.exception_response(response_code, response_dict.get('errorSummary', ''))
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def get_token(self):
```

Comment:
Should this method be private? It's only used internally to this class and it only really functions with one of the other calls right?

Reply:
Updated as per the suggestion.